app/helpers/authentication_helpers.rb

```diff
@@ -16,18 +16,27 @@ def warden
   # Reads details from the params fetched from the caller context.
   #
   def authenticated?
+    # Variable to store auth_token if available
+    token = nil
     # Check warden -- authenticate using DB or LDAP etc.
     return true if warden.authenticated?
-
-    # Check for auth token parameter
-    if params.present? && params[:auth_token].present?
-      # Get the token and the user - if there is a token
-      token = AuthToken.find_by_auth_token(params[:auth_token])
-      user_by_token = token.user unless token.nil?
+    # Check for valid auth token  and username in request header 
+    if headers.present? && headers['Auth-Token'].present? && headers['Username'].present? && User.find_by_username(headers['Username']).present?
+      # Get the list of tokens for a user 
+      valid_user = User.find_by_username(headers['Username']) #user_by_token
+      tokens_by_user = AuthToken.where(user_id: valid_user.id) #token
```

Comment:
Move this into the user model - then you can get just the tokens associated with that user. (No need to user the class directly, you can get them via the relation).

Reply:
Maybe call it `token_with_value` and have it return the matching token or nil